### PR TITLE
Python 3 Upgrade [Re-try]

### LINKED ISSUE #340: Update Docs (Part 2 & 3)
Part 2
- [x] Update references in index and README
- [x] Fix wording errors and deprecated references

Part 3
- [x] Add documentation in tasks.py (for "Scheduled Tasks" section of "Data Pipeline Documentation")
- [x] Fix bugs in `environment.rst` and review bugs from parser as noted when you build
- [x] Change `readthedocs` link in `readme` to https

### LINKED ISSUE #343: Upgrade Sphinx (Readthedocs) to Python 3
**Issue**
Python 2 is already deprecated and we're already close to upgrading to Python 3. Once it's completed, readthedocs should reflect these changes.

**Solution**
Update `.readthedocs.yaml` with a simple change to version 3.x
If the readthedocs build errors, then adjust accordingly

### PR DESCRIPTION
**Pull request by Boccca2014**
resolves #317 
resolves #343 
resolves #340

TODO-
- [x] Update semesterly-base (Docker) build and replace sgerli's version with ours
       - "FROM sgerli/horariotec-base:latest" to "FROM scabrej1/semesterly-py3-base:latest"
- [x] Change from Python 2.7 to 3.8.5
- [x] Fix travis failures
- [x] HIGHLY recommend updating multiple dependencies when completing this task (noted in #320)
- [x] Update docker image
- [x] Resolve migrations
- [x] Copy over `scabrej1` ([linked here](https://hub.docker.com/r/scabrej1/semesterly-py3_8-base/tags?page=1&ordering=last_updated)) docker image to `jhuopensource` @alihacks 

Next steps:
- check how this will impact the "cs310-spring21" branch
- Update npm dependencies

Resolved Issues:
- Fails travis because issue with numpy version currently
- Migrations require merge, but merging breaks database

Solutions:
- Try other method with "apt-get key" on [stackoverflow ](https://askubuntu.com/questions/766883/there-is-no-public-key-available-for-the-following-key-ids-1397bc53640db551)
- Clean out ALL migrations for student and timetable models and redo all migrations

Error log:
* **You are likely to encounter this error**: ImportError: bad magic number
  * This is caused by outdated compiled python files. On unix systems you can run `(sudo) find . -name '*.pyc' -delete`
* Migration tree-dependency error
  * Caused by having two migrations that stem off from the same parent migration, which occurred because development continued after the migration to python3 occurred. **This may cause issues** because any code written after Michael might still be python3 incompatible.
* Database timezone not in UTC
  * Caused by psycopg2-binary being version 2.9.1 instead of 2.8.6, fixed this by editing the Dockerfile.
* NameError: name 'basestring' is not defined
  * 'basestring' was the base class between str and bytes in python2, but now python3 just uses 'str' instead.

James 
* Resolved errors in error log above ^
* Resolved migration errors
* Resolved bugs in courses.py, jsonschema, & logger.py -> now ingests and digests data locally

### CODE REVIEW

**Comment by Boccca2014** [comment]
[Possible solution](https://askubuntu.com/questions/766883/there-is-no-public-key-available-for-the-following-key-ids-1397bc53640db551) to travis failure

**Comment by Boccca2014** [comment]
Note: May need to clean out all migration files in student and timetable to overcome problems with overlapping fields.

File: .travis.yml [hunk 1]
```diff
@@ -1,14 +1,17 @@
 os: linux
 sudo: required
-dist: xenial
-services:
-  - xvfb
-  - postgresql
-language: python
+dist: trusty
```
Comment (alihacks) [inline .travis.yml:3]: why are we going back on the linux version? I believe this is the cause of the build fail

File: Dockerfile [hunk 2]
```diff
@@ -1,5 +1,5 @@
-FROM jhuopensource/semesterly-base:latest
-
+FROM sgerli/horariotec-base:latest
```
Comment (alihacks) [inline Dockerfile:1]: we should not be using their base as changes there can break our prod build in the future. To make the build pass, I think you should use your own account like you were, once you have a working base image, I'd be happy to push that to jhuopensource but as a different image so we don't break the 2.x builds, paybe semesterly-py3-base

**Review by alihacks — APPROVED** [review]

**Pull request merged**